src/expression/function/evaluate.js

```diff
@@ -1,9 +1,18 @@
 import { deepMap } from '../../utils/collection.js'
 import { factory } from '../../utils/factory.js'
+import { isMapLike } from '../../utils/customs.js'
 
 const name = 'evaluate'
 const dependencies = ['typed', 'parse']
 
+function deprecatePlainObjectScope (scope) {
+  // Using bare objects as scopes is a source of potential security vulnerabilities.
+  // https://github.com/josdejong/mathjs/issues/2165
+  if (!isMapLike(scope)) {
+    console.warn('⚠️ Using bare objects as scopes is deprecated and may be removed in future versions. Use Map instead. ⚠️')
+  }
```

Comment:
An alternative here would be wrap the object in a `Map`, and then stop using `[g|s]etSafeProperty` for scopes entirely.

I didn't do it here, yet, because I don't know want to break APIs.